pkg/cloudprovider/aws/amifamily/bootstrap/bottlerocketsettings.go

```diff
@@ -0,0 +1,98 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+// config is the root of the bottlerocket config, see more here https://github.com/bottlerocket-os/bottlerocket#using-user-data
+type config struct {
+	Settings settings `toml:"settings"`
+}
+
+// This is a subset of all configuration in https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/models/src/aws-k8s-1.22/mod.rs
+// These settings apply across all K8s versions that karpenter supports.
+// This is currently an opinionated subset and can evolve over time
+type settings struct {
+	Kubernetes     kubernetes      `toml:"kubernetes"`
+	HostContainers *hostContainers `toml:"host-containers,omitempty"`
+	AWS            *awsConfig      `toml:"aws,omitempty"`
+	Metrics        *metrics        `toml:"metrics,omitempty"`
+	Kernel         *kernel         `toml:"kernel,omitempty"`
+}
+
+// kubernetes specific configuration for bottlerocket api
+type kubernetes struct {
+	APIServer                 string               `toml:"api-server"`
+	ClusterCertificate        *string              `toml:"cluster-certificate"`
+	ClusterName               *string              `toml:"cluster-name"`
+	ClusterDNSIP              *string              `toml:"cluster-dns-ip,omitempty"`
+	NodeLabels                map[string]string    `toml:"node-labels,omitempty"`
+	NodeTaints                map[string][]string  `toml:"node-taints,omitempty"`
+	MaxPods                   *int                 `toml:"max-pods,omitempty"`
+	StaticPods                map[string]staticPod `toml:"static-pods,omitempty"`
+	EvictionHard              map[string]string    `toml:"eviction-hard,omitempty"`
+	KubeReserved              map[string]string    `toml:"kube-reserved,omitempty"`
+	SystemReserved            map[string]string    `toml:"system-reserved,omitempty"`
```

Comment:
should a lot of these be added to the Provisioner spec.kubeletConfig and then merged into the bottlerocket config?